packages/nextjs/src/config/loaders/ast.ts

```diff
@@ -320,3 +328,169 @@ export function removeComments(ast: AST): void {
+  const hasDefaultDeclaration = ast.find(ExportDefaultDeclaration).size() > 0;
+  if (hasDefaultDeclaration) {
+    return true;
+  }
+
+  const hasDefaultSpecifier = ast.find(ExportDefaultSpecifier).size() > 0;
+  if (hasDefaultSpecifier) {
+    return true;
+  }
+
+  const hasNamedDefaultExport = ast.find(ExportSpecifier).some(specifier => specifier.node.exported.name === 'default');
+  if (hasNamedDefaultExport) {
+    return true;
+  }
+
+  return false;
+}
+
+/**
+ * Extracts all identifiers from a "RestElement" within a named export declaration statement ("export const val = ...").
+ * "RestElements" are things like "...value" inside a destructuring assignment.
+ *
+ * Example:
+ * "export const { ...restElement1, bar: { ...restElement2 } } = { foo: 1, bar: { baz: 2 } }" --> ["restElement1", "restElement2"]
+ */
+function getExportIdentifiersFromRestElement(restElement: jscsTypes.RestElement): string[] {
+  const identifiers: string[] = [];
+
+  if (Identifier.check(restElement.argument)) {
+    identifiers.push(restElement.argument.name);
+  }
+
+  return identifiers;
+}
+
+/**
+ * Extracts all identifiers from an ArrayPattern within a destructured named export declaration
+ * statement ("export const [val] = [1]").
+ *
+ * This function recursively calls itself and `getExportIdentifiersFromObjectPattern` since destructuring assignments
+ * can be deeply nested with objects and arrays.
+ *
+ * Example:
+ * export const [{ foo: name1 }, [{ bar: [name2]}, name3]] = [{ foo: 1 }, [{ bar: [2] }, 3]];
+ *   --> ["name1", "name2", "name3"]
+ */
+function getExportIdentifiersFromArrayPattern(arrayPattern: jscsTypes.ArrayPattern): string[] {
+  const identifiers: string[] = [];
+
+  arrayPattern.elements.forEach(element => {
+    if (Identifier.check(element)) {
+      identifiers.push(element.name);
+    } else if (ObjectPattern.check(element)) {
+      identifiers.push(...getExportIdentifiersFromObjectPattern(element));
+    } else if (ArrayPattern.check(element)) {
+      identifiers.push(...getExportIdentifiersFromArrayPattern(element));
+    } else if (RestElement.check(element)) {
+      identifiers.push(...getExportIdentifiersFromRestElement(element));
+    }
+  });
+
+  return identifiers;
+}
+
+/**
+ * Grabs all identifiers from an ObjectPattern within a destructured named export declaration
+ * statement ("export const { val: name } = { val: 1 }").
+ *
+ * This function recursively calls itself and `getExportIdentifiersFromArrayPattern` since destructuring assignments
+ * can be deeply nested with objects and arrays.
+ *
+ * Example:
+ * export const { foo: [name1], bar: { baz: [{ quux: name2 }], ...name3 }} = { foo: [1], bar: { baz: [{ quux: 3 }]} };
+ *   --> ["name1", "name2", "name3"]
+ */
+function getExportIdentifiersFromObjectPattern(objectPatternNode: jscsTypes.ObjectPattern): string[] {
+  const identifiers: string[] = [];
+
+  objectPatternNode.properties.forEach(property => {
+    if (Property.check(property)) {
```

Comment:
If you read to the end of the function, and correctly parse the nested conditionals, this becomes obvious, but a comment here forestalls the, "Wait, they're properties! How could they not be `Property`s?" confusion, and, as a bonus, is a subtle hint that the last `else if` goes with this `if`, not any of the inner stuff.

```suggestion
    // An `ObjectPattern`'s properties can be either `Property`s or `RestElement`s.
    if (Property.check(property)) {
```